packages/bot-web-ui/src/components/transactions/__tests__/transaction.spec.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { mockStore, StoreProvider } from '@deriv/stores';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { render, screen } from '@testing-library/react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import userEvent from '@testing-library/user-event';
+import { mock_ws } from 'Utils/mock';
+import RootStore from 'Stores/root-store';
+import { DBotStoreProvider, mockDBotStore } from 'Stores/useDBotStore';
+import Transaction from '../transaction';
+
+jest.mock('@deriv/bot-skeleton/src/scratch/blockly', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => ({}));
+jest.mock('@deriv/bot-skeleton/src/scratch/hooks/block_svg', () => jest.fn());
+
+const mock_contract = {
+    transaction_ids: { buy: '12345', sell: '6789' },
+    underlying: 'EURUSD',
+    entry_tick: 1.2345,
+    exit_tick: 1.6789,
+    entry_tick_time: '5pm',
+    exit_tick_time: '6pm',
+    date_start: 10,
+    tick_count: 100,
+    buy_price: 50,
+    currency: 'USD',
+    profit: 30,
+    barrier: 3,
+    is_completed: true,
+};
+
+describe('Transaction', () => {
+    let wrapper: ({ children }: { children: JSX.Element }) => JSX.Element, mock_DBot_store: RootStore | undefined;
+
+    beforeEach(() => {
+        const mock_store = mockStore({});
+        mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+
+        wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock_store}>
+                <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
+                    {children}
+                </DBotStoreProvider>
+            </StoreProvider>
+        );
+    });
+
+    it('should render the Transaction component', () => {
+        const { container } = render(<Transaction />, { wrapper });
+        expect(container).toBeInTheDocument();
+    });
+
+    it('should render transaction details when a contract is provided', () => {
+        render(<Transaction contract={mock_contract} />, { wrapper });
+
+        expect(screen.getByText('1.2345')).toBeInTheDocument();
+        expect(screen.getByText('1.6789')).toBeInTheDocument();
+        expect(screen.getByText('50.00 USD')).toBeInTheDocument();
+        expect(screen.getByText('30.00 USD')).toBeInTheDocument();
+    });
+
+    it('should render the Transaction component and trigger onClick', () => {
+        const setActiveTransactionIdMock = jest.fn();
+
+        render(<Transaction contract={mock_contract} setActiveTransactionId={setActiveTransactionIdMock} />, {
+            wrapper,
+        });
+
+        const transaction_item = screen.getByTestId('dt-transactions-item');
+        userEvent.click(transaction_item);
```

Comment:
here you may handle keyDown event too, that is expected

import { render, screen, **fireEvent** } from '@testing-library/react';
```suggestion
        userEvent.click(transaction_item);
        
        fireEvent.keyDown(transaction_item, { key: 'Enter' });
```
<img width="925" alt="Screenshot 2024-02-21 at 12 17 51" src="https://github.com/binary-com/deriv-app/assets/103181650/20644e6f-6133-45e0-8829-9bf4b86363a1">